### ISSUE TITLE
🤖 [AUTO] Implement Strategy: boosted_regression_tree_spy

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#884]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 6)
**Extracted:** 2026-01-11T07:22:14.079122

---

1.  **Strategy Name**: boosted_regression_tree_spy
2.  **Strategy Type**: Prediction-based
3.  **Timeframe**: 1D
4.  **Instruments**: SPY (S&P 500 ETF)

5.  **Entry Rules**:
    *   **Long**: Predicted future one-day return (`retPred1`) from the final boosted model is positive.
    *   **Short**: Predicted future one-day return (`retPred1`) from the final boosted model is negative.

6.  **Exit Rules**:
    *   **Take Profit**: Not explicitly defined; assumed to hold for one day.
    *   **Stop Loss**: Not explicitly defined; assumed to hold for one day.
    *   **Time-based Exit**: Hold positions for exactly one day.

7.  **Indicators Required**:
    *   Previous 1-day return (`ret1`)
    *   Previous 2-day return (`ret2`)
    *   Previous 5-day return (`ret5`)
    *   Previous 20-day return (`ret20`)
    *   **Model**: Regression Tree (as the base learner).
    *   **Ensemble Technique**: Boosting (e.g., MATLAB's `fitensemble` function with `LSBoost` method and `Tree` learner).
    *   **Hyperparameter**: `M` (number of boosting iterations).

8.  **Risk Management**:
    *   **Position Sizing**: Buy SPY with a fixed dollar amount for long signals, short SPY with a fixed dollar amount for short signals.
    *   **Maximum Risk per Trade**: Not explicitly defined.
    *   **Risk-Reward Ratio**: Not explicitly defined.

9.  **Additional Notes**:
    *   Boosting iteratively applies a learning algorithm to the prediction errors of the model from the previous iteration.
    *   In the provided example, boosting did not significantly improve out-of-sample performance, and the Sharpe ratio remained insignificant, suggesting it did not alleviate overfitting for this particular problem setup.

---

## Implementation Instructions

Please implement this strategy as a **MoonDevStrategy plugin** with the following requirements:

1. **File Location:** `strategies/boosted_regression_tree_spy.py`
2. **Inheritance:** Your class MUST inherit from `MoonDevStrategy`:
   ```python
   from src.strategies.base import MoonDevStrategy
   
   class MyStrategy(MoonDevStrategy):
       def init(self):
           # Initialize indicators
           pass
           
       def next(self):
           # Trading logic
           pass
   ```
3. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting.

### ⚠️ IMPORTANT: VuManchu / Market Cipher / Cipher B Indicators

This repository has a **pre-built VuManchu indicator library**. You MUST use it:

```python
# In your preprocess_data function:
from src.indicators.vumanchu import cipher_b, wavetrend, rsimfi

def preprocess_data(df, **params):
    df = cipher_b(df)  # Adds: wt1, wt2, buy_signal, sell_signal, rsimfi, stoch_rsi_k, stoch_rsi_d
    return df
```

**Available functions in `src.indicators.vumanchu`:**
- `cipher_b(df)` - Complete Cipher B suite (WaveTrend + MFI + StochRSI)
- `wavetrend(df)` - WaveTrend oscillator only (wt1, wt2, buy_signal, sell_signal)
- `rsimfi(df)` - RSI + Money Flow Index composite
- `stoch_rsi(df)` - Stochastic RSI

**Reference implementation:** See `strategies/vumanchu_cipher_b.py` for a working example.

### 📋 Strategy Development Guidelines

**CRITICAL:** Your implementation MUST follow the guidelines in `.agent/rules/strategy_development.md`. Key requirements:

1. **ATR-Based Risk Management:**
   - Stop loss: `entry ± (2 * ATR)`
   - Take profit: `entry ± (3+ * ATR)`

2. **Multi-Timeframe Filter:**
   - Require higher TF trend confirmation (4H minimum)

3. **Volume Confirmation:**
   - Require volume > average for entries

4. **No Hard-Coded Values:**
   - ❌ Fixed time windows (e.g., 10-11 AM)
   - ❌ Fixed price targets (e.g., +5 points)
   - ✅ Adaptive based on ATR/volatility

**Validation:** All strategies must pass Walk-Forward Analysis. Failure to follow guidelines = failed WFA.

3. **Other Indicators:** For standard indicators, use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators).
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
